bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/ReadEntryProcessor.java

```diff
@@ -122,6 +122,7 @@ protected void processPacket() {
         if (LOG.isTraceEnabled()) {
             LOG.trace("Read entry rc = {} for {}", errorCode, request);
         }
+        request.recycle();
```

Comment:
sendResponse still using request. Maybe we should recycle at the ReadEntryProcessor#recycle()?

Reply:
Yes, we can only recycle the request here.